### PR TITLE
[mock_uss] Modify sharing behavior in mock_uss on injection

### PR DESCRIPTION
**Pull request by punamverma**
This PR adds the ability to modify sharing behavior of mock_uss on injection, as would be needed by negative tests for data validation of SUT.
1. Created MockUssInjectFlightRequest (subclass of InjectFlightRequest) that enables adding MockUssFlightBehavior for modifying fields while sharing ops using GET or POST request.
2. Handled the MockUssFlightBehavior in mock uss routes_injection and routes_scdsc to share modified op in GET and POST requests. DSS has the correct operational_intent, its the operational_intent shared with USSes that is modified for the purpose of writing data validation tests.
3. Added the ability to inject MockUssFlightBehavior through flight_planner's request_flight method.

### CODE REVIEW

File: monitoring/uss_qualifier/scenarios/flight_planning/test_steps.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import inspect
 from typing import List, Optional, Tuple, Iterable, Set, Dict, Union
+import time
+from datetime import datetime
```
Comment (BenjaminPelletier) [inline monitoring/uss_qualifier/scenarios/flight_planning/test_steps.py:4]: These imports appear unnecessary

File: monitoring/monitorlib/clients/flight_planning/client_scd.py [hunk 2]
```diff
@@ -45,6 +49,7 @@ def _inject(
         flight_id: FlightID,
         flight_info: FlightInfo,
         execution_style: ExecutionStyle,
+        mod_flight_behavior: Optional[MockUssFlightBehavior] = None,
```
Comment (BenjaminPelletier) [inline monitoring/monitorlib/clients/flight_planning/client_scd.py:52]: This client is a particular implementation of the generic client -- we shouldn't change method signatures as that will prevent the use of other implementations of the generic client.  Instead, it seems reasonable to add an `additional_fields: Optional[dict] = None` argument to this function in FlightPlannerClient (and therefore here as well) whose fields get added to the request body just before it's dispatched to the remote server.  Doing it this way will allow easy adaptation for other implementations of FlightPlannerClient of which there will soon be one more (for flight_planner interface).

**Review by BenjaminPelletier — COMMENTED** [review]
I have to get to bed; more comments tomorrow, but here's what I have so far.

**Comment by punamverma** [comment]
> I have to get to bed; more comments tomorrow, but here's what I have so far.

Thanks for the comments. Sorry, didn't mean to disturb you.

**Comment by BenjaminPelletier** [comment]
> > I have to get to bed; more comments tomorrow, but here's what I have so far.
> 
> Thanks for the comments. Sorry, didn't mean to disturb you.

Oh, no, not at all -- I only review when I'm willing to work.  Please feel free to push/comment/update/message/etc at any time regardless of time of day.  I just usually like providing a complete set of comments, so wanted to explain why this set was not complete (seemed useful to get these out sooner rather than waiting for a complete set)

File: monitoring/uss_qualifier/scenarios/flight_planning/test_steps.py [hunk 3]
```diff
@@ -92,7 +97,7 @@ def plan_flight_intent(
     scenario: TestScenarioType,
     test_step: str,
     flight_planner: FlightPlanner,
-    flight_intent: InjectFlightRequest,
+    flight_intent: MockUssInjectFlightRequest,
```
[the remark this answers is not in the document]
Comment (punamverma) [reply monitoring/uss_qualifier/scenarios/flight_planning/test_steps.py]: Changed.

File: monitoring/uss_qualifier/scenarios/flight_planning/test_steps.py (hunk 1 above)
Comment (punamverma) [reply monitoring/uss_qualifier/scenarios/flight_planning/test_steps.py]: Removed.

File: monitoring/monitorlib/mock_uss_interface/mock_uss_scd_injection_api.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+from implicitdict import ImplicitDict
+from typing import List, Optional
+from uas_standards.interuss.automated_testing.scd.v1.api import InjectFlightRequest
+
+
+class MockUssFlightBehavior(ImplicitDict):
+    modify_sharing_methods: List[str]
+    modify_fields: dict
```
[the remark this answers is not in the document]
Comment (punamverma) [reply monitoring/monitorlib/mock_uss_interface/mock_uss_scd_injection_api.py]: Added documentation.

File: monitoring/monitorlib/clients/flight_planning/client_scd.py (hunk 2 above)
Comment (punamverma) [reply monitoring/monitorlib/clients/flight_planning/client_scd.py]: Added additional_fields.

**Comment by punamverma** [comment]
@BenjaminPelletier I have made changes using additional_fields. Could you please check. Thanks.

**Comment by BenjaminPelletier** [comment]
I can help with the merge on this one -- I'll make a PR to your branch

**Review by BenjaminPelletier — COMMENTED** [review]
I'll propose a pull request to this branch that both merges main and addresses these comments

File: monitoring/mock_uss/scdsc/routes_scdsc.py [hunk 5]
```diff
@@ -39,21 +45,44 @@ def scdsc_get_operational_intent_details(entityid: str):
         )
 
     # Return nominal response with details
-    response = GetOperationalIntentDetailsResponse(
-        operational_intent=op_intent_from_flightrecord(flight),
-    )
+    response = {"operational_intent": op_intent_from_flightrecord(flight)}
```
[the remark this answers is not in the document]
Comment (punamverma) [reply monitoring/mock_uss/scdsc/routes_scdsc.py:48]: I think I had tried that. But, that lead to validation of the operational_intent as per GetOperationalIntentDetailsResponse. Whereas, we want to have invalid data in our test.

[the remark this answers is not in the document]
Comment (BenjaminPelletier) [reply monitoring/mock_uss/scdsc/routes_scdsc.py:48]: Types are validated upon parsing, but not upon construction -- an invalid value can be provided upon construction.

File: monitoring/mock_uss/scdsc/routes_scdsc.py [hunk 6]
```diff
@@ -39,21 +45,44 @@ def scdsc_get_operational_intent_details(entityid: str):
...
+    method = "GET"
+    if "mod_op_sharing_behavior" in flight:
+        mod_op_sharing_behavior = ImplicitDict.parse(
+            flight.mod_op_sharing_behavior, MockUssFlightBehavior
+        )
+        if mod_op_sharing_behavior.modify_sharing_methods is not None:
+            if method not in mod_op_sharing_behavior.modify_sharing_methods:
+                return OperationalIntent(reference=ref, details=details)
+        if mod_op_sharing_behavior.modify_fields is not None:
+            if "operational_intent_reference" in mod_op_sharing_behavior.modify_fields:
```
[the remark this answers is not in the document]
Comment (BenjaminPelletier) [reply monitoring/mock_uss/scdsc/routes_scdsc.py:69]: Why would having modify_fields apply directly to OperationalIntent cause validation versus applying to the two member separately?

[the remark this answers is not in the document]
Comment (punamverma) [reply monitoring/mock_uss/scdsc/routes_scdsc.py:69]: Let me pull and check, as while running my tests, I had come across validation errors. But, let me pull your PR and check. Could have been some issue with my code. Will check and let you know if I see any issues. Thanks

**Review by BenjaminPelletier — APPROVED** [review]

**Pull request merged**